pytest_elk_reporter.py

```diff
@@ -192,8 +205,13 @@ def __init__(self, config):
         self.is_slave = False
 
     @property
-    def es_auth(self):
-        return self.es_username, self.es_password
+    def es_auth_args(self) -> dict[str, Any]:
+        output = {}
+        if self.es_api_key:
+            output.update(dict(headers={"Authorization": f"ApiKey {self.es_api_key}"}))
```

Comment:
probably we can just return `output` here? IMHO, API key is more secure than username and password. if that's the case, we could just put
```py
if self.es_api_key:
    headers = {"Authorization": f"ApiKey {self.es_api_key}"}
    return {"headers": headers}
```

Reply:
that was I did in the beginning, pylint wasn't happy about multiple returns within if/else

yes we can end up with both headers, don't know what would be the behavior of ES in this case

but I'll test it

didn't yet integrated with dtest, I'll try this to see if it work, or we need to prevent user for supplying both key and user/pass

Reply:
I've change back, after testing, if both header passed down, and one is wrong you fail to connect
so api key comes first, if both are defined.

raising exception if both set, in `__init__` would be enough, cause in most usages we change those on runtime, after we retrieve the credentials, and never passing them with command line or configuration file.

I think that's good enough, and no need to go overboard with guardrailing this.